wordpress-mobile/WordPress-iOS
# Auth: Updates so an account can be created with just an authtoken.

This PR updates WPiOS to accommodate changes in WordPressAuthenticator-iOS that remove the `username` field from the credentials enum.  These changes allow an account to be created with _just_ an auth token.  Taken together it will allow a user to use magic links requested from outside of the mobile app (e.g. Calypso, wp-admin, etc.). 

To test:
- Test the wpcom login flows.  Ensure you can log in with an email address + password, and by requesting a magic link.  Ensure that no information is missing from the epilogue. 
- Shut down the simulator then relaunch it.
- Confirm you are still logged in in the Me tab.

Bonus: _After_ requesting a magic link uninstall and reinstall the app, then try the magic link. It _should_ log you in correctly. 

Needs review: @ScoutHarris here is the companion to the WordPressAuthentication-iOS PR.

I've opened this as a draft since the Podfile is pointing to a commit hash. I'll update it once the auth pod PR lands.
